Source/BSGConnectivity.m

```diff
@@ -50,21 +50,19 @@ @interface BSGConnectivity ()
 
 @property(nonatomic, assign) SCNetworkReachabilityRef reachabilityRef;
 @property(nonatomic, strong) dispatch_queue_t serialQueue;
-@property(nonatomic) CallbackBlock callbackBlock;
+@property(nonatomic, copy) CallbackBlock callbackBlock;
 
 @end
 
 @implementation BSGConnectivity
 
 - (instancetype)initWithURL:(NSURL *)url
                 changeBlock:(ConnectivityChange)changeBlock {
-    NSString *hostName = [url absoluteString];
-    SCNetworkReachabilityRef ref =
```

Comment:
In the case that `[super init]` returns nil, the `ref` would never be released and thus cause a leak.